packages/ethereum/deploy/08_HoprStake.ts

```diff
@@ -28,7 +27,11 @@ const main: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   await deploy('HoprStake', {
     contract: stakeContractName,
     from: deployer,
-    args: [HoprBoost.address, admin, xHOPR.address, wxHOPR.address],
+    // beyond staking season 4 (from S5 on, no need to pass constructor parameters)
+    args:
+      parseInt([...stakeContractName.matchAll(/HoprStakeSeason(\d+)/g)][0][1], 10) > 4
```

Comment:
Let's move this into an assignment before this function call to make it easier to read.